sdk/communication/communication-short-codes/src/shortCodesClient.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/// <reference lib="esnext.asynciterable" />
+
+import {
+  parseClientArguments,
+  isKeyCredential,
+  createCommunicationAuthPolicy
+} from "@azure/communication-common";
+import { isTokenCredential, KeyCredential, TokenCredential } from "@azure/core-auth";
+import {
+  PipelineOptions,
+  InternalPipelineOptions,
+  createPipelineFromOptions,
+  RestResponse
+} from "@azure/core-http";
+import { PagedAsyncIterableIterator } from "@azure/core-paging";
+import { logger, createSpan, SDK_VERSION } from "./utils";
+import { ShortCodesClient as ShortCodesGeneratedClient } from "./generated/src";
+import { ShortCodes as GeneratedClient } from "./generated/src/operations";
+import {
+  ShortCode,
+  ShortCodesGetShortCodesOptionalParams,
+  ShortCodesUpsertUSProgramBriefOptionalParams,
+  USProgramBrief
+} from "./generated/src/models/";
+import { SpanStatusCode } from "@azure/core-tracing";
+import {
+  DeleteUSProgramBriefOptions,
+  GetUSProgramBriefOptions,
+  SubmitUSProgramBriefOptions
+} from "./models";
+
+/**
+ * Client options used to configure the ShortCodesClient API requests.
+ */
+export interface ShortCodesClientOptions extends PipelineOptions {}
+
+const isShortCodesClientOptions = (options: any): options is ShortCodesClientOptions =>
+  options && !isKeyCredential(options) && !isTokenCredential(options);
+
+export class ShortCodesClient {
+  /**
+   * A reference to the auto-generated ShortCodes HTTP client.
+   */
+  private readonly client: GeneratedClient;
+
+  public constructor(connectionString: string, options?: ShortCodesClientOptions);
+
+  public constructor(url: string, credential: KeyCredential, options?: ShortCodesClientOptions);
```

Comment:
I think we're calling the first argument `endpoint` in our other ACS libraries. Let's rename here and on line 52.